torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2605,40 +2605,41 @@ def _register_pre_backward_hooks(self, outputs: Any) -> Any:
         self._pre_backward_hook_has_run = False
 
         def _pre_backward_hook(*unused: Any) -> None:
-            # Run ``_pre_backward_hook`` only once per backward pass
-            if self._pre_backward_hook_has_run:
-                return
-            # try to queue final backward callback only once for root, so
-            # that final backward callback is attached to the outer most
-            # backward graph task and called after all the backward
-            # calls are completed.
-            if self._is_root:
-                self._queue_wait_for_post_backward()
-
-            if self._need_prefetch_pre_backward_hook():
-                # Always wait for all_gather before rebuilding full params, just
-                # in case full params have already been prefetched in previous layer's
-                # pre-backward hook.
+            with torch.autograd.profiler.record_function("FullyShardedDataParallel._pre_backward_hooks"):
+                # Run ``_pre_backward_hook`` only once per backward pass
+                if self._pre_backward_hook_has_run:
+                    return
```

Comment:
We might want to enter the record_function context after this. _pre_backward_hook can run many times (as it is a hook registered on tensors), but it only does actual work once, so let's profile just that.

Reply:
Done, thanks for the suggestion!